Migrate build system to net8 / fake  v6

anpin opened the pull request:
This PR updates build scripts to FAKE v6 and updates test targets to `net8.0` thus reducing the need for developer to have multiple ~outdated~ build tools to get started. 

Fixes 
#360
#362

## Changes

- Updated FAKE from v4 to v6 and removed old bootstrap scripts 
- `NetCoreTestVersion` updated to `net8.0` (which required `MySql.Data` to be updated to latest version)
- DocFx updated to latest version which uses `dotnet-tools` (allows working with docs on mac/linux) 
- DocFx serve is invoked as a part  of `build.fsx` and `serve-docs.[cmd|sh]` are updated to use that
- Fixed broken links in docs
- Refactored docker container initialization to fix inconsistent test runs
- Adds benchmarks to the build script

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed, if any.
* [ ] I have added website documentation for this feature.

### Latest `dev` Benchmarks 

Include data from the [relevant benchmark](https://getakka.net/community/contributing/index.html#improve-performance) prior to this change here.

### This PR's Benchmarks

Include data from after this change here.

anpin committed:
migrate build system to net8 / fake  v6

anpin commented:
currently runs on linux, windows not tested yet. Plus there are some errors in the tests themselves, probably due to mismatched dependencies.

anpin commented:
tests seem to be rather flaky both on my machine and in CI. containers sometimes fail on startup.

anpin committed:
run benchmarks from build script

anpin committed:
don't run benchmarks on windows

Aaronontheweb commented:
@anpin we definitely have some flaky tests - I'll try to take a look at this. Thank you for your hard work

anpin commented:
@Aaronontheweb my apologies for not discussing this major change upfront, but I need to build the package from source and didn't want to use some temporary scripts

anpin commented:
regarding failing test https://github.com/akkadotnet/Akka.Persistence.Sql/issues/362

anpin committed:
fix execution order

anpin committed:
increase timeout for benchmarks

anpin committed:
refactored docker container initialization
- previous GetContainerLogsAsync method returning stream was marked obsolete, so I replaced it with the IProgress based version
- Added retry loop in case container fails to start (e.g. port is alraedy in use)

anpin commented:
Let's see if tests would run consistently in CI now. On my machine they do not fail randomly with last commit.

anpin committed:
update release notes

anpin commented:
updating release notes just to trigger the build again and see if any error would occur

anpin commented:
seems to be working. @Aaronontheweb please let me know if anything else is required to merge this

Aaronontheweb commented on .config/dotnet-tools.json:
```diff
@@ -7,6 +7,12 @@
         "jb"
       ]
+    },
+    "docfx": {
```
WE can probably get rid of DocFx altogether (it was just part of the template for this repo) but I appreciate you doing this

anpin replied:
Will probably make CI faster. I figured it could be executed from some over place where the documentation is built, so I tried to keep it comparability reasons

Aaronontheweb commented on build-system/azure-pipeline.template.yaml:
LGTM

Aaronontheweb commented on build-system/windows-release.yaml:
LGTM

Aaronontheweb commented on build.fsx:
LGTM

Aaronontheweb commented on build.sh:
LGTM - much better lol

Aaronontheweb commented on docs/docfx.json:
```diff
@@ -62,7 +62,7 @@
         }],
         "xref": [
-            "../tools/msdn.4.5.2/content/msdn.4.5.2.zip"
+            "https://learn.microsoft.com/en-us/dotnet/.xrefmap.json"
```
We should do this in the main Akka.NET repo as well - uses the same system (I would love to get off of the old FAKE there too)

Aaronontheweb commented on src/Akka.Persistence.Sql.Data.Compatibility.Tests/Internal/DockerContainer.cs:
LGTM

Aaronontheweb reviewed (approved):
LGTM - nice work. Would love it if we could bring some of this stuff to the main Akka.NET project too (it also uses an aging FAKE system)

Aaronontheweb commented:
lol this reminds me, I need to finish getting https://github.com/akkadotnet/akka.net/pull/7063 merged in

anpin commented:
> LGTM - nice work. Would love it if we could bring some of this stuff to the main Akka.NET project too (it also uses an aging FAKE system)

Thanks. I can probable take a look at the main repo too.
